scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Record.java

```diff
@@ -330,16 +342,16 @@ private Record(ProtocolVersion version, int epoch, long sequenceNumber) {
 			if (type == ContentType.TLS12_CID.getCode()) {
 				if (cidGenerator == null) {
 					LOGGER.debug("Received TLS_CID record, but cid is not supported. Discarding ...");
-					continue;
+					return records;
 				} else if (cidGenerator.useConnectionId()) {
 					connectionId = cidGenerator.read(reader);
 					if (connectionId == null) {
 						LOGGER.debug("Received TLS_CID record, but cid is not matching. Discarding ...");
-						continue;
+						return records;
 					}
 				} else {
 					LOGGER.debug("Received TLS_CID record, but cid is not used. Discarding ...");
-					continue;
+					return records;
```

Comment:
was it a bug ?

Reply:
It's more a optimization.
Failing to read the record to the end will fail the next record as well. So the result will close to almost be the same.